### PR TITLE
Features/confirms for router navigate

### PR DESCRIPTION
**Pull request by eugene-sahach**

### CODE REVIEW

File: src/Uintra20/Content/uintra20/src/app/feature/project/specific/activity/news-form/news-form.component.ts [hunk 1]
```diff
@@ -131,6 +148,11 @@ export class NewsFormComponent implements OnInit {
     }
   }
 
+  //Main cancel function
+  onCancel() {
+    this.router.navigate(['/']);
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/feature/project/specific/activity/news-form/news-form.component.ts]: use `this.handleCancel.emit();` and remove hardcoded link

File: src/Uintra20/Content/uintra20/src/app/feature/project/specific/activity/news-form/news-form.component.ts [hunk 2]
```diff
@@ -151,7 +173,24 @@ export class NewsFormComponent implements OnInit {
   }
 
   changeOwner(e) {
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/feature/project/specific/activity/news-form/news-form.component.ts]: rename e. it is not event

Comment (razrabotal) [reply src/Uintra20/Content/uintra20/src/app/feature/project/specific/activity/news-form/news-form.component.ts]: maybe add interface

File: src/Uintra20/Content/uintra20/src/app/ui/pages/article/article-page.html [hunk 3]
```diff
@@ -1,3 +1,12 @@
...
+  *ngFor="let panel of data.panels.get()"
+  [data]="panel.get()"
+>
+</ubl-dynamic-component>
+
+<div *ngIf="isShow" class="confirm">
+  <div class="confirm-content">
+  <button (click)="onSubmit()">Ok</button>
+  <button (click)="onCancel()">Not</button>
+</div>
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/ui/pages/article/article-page.html]: remove it

File: src/Uintra20/Content/uintra20/src/app/ui/pages/article/article-page.less [hunk 4]
```diff
@@ -2,4 +2,23 @@
...
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+.confirm-content {
+  width: 400px;
+  height: 400px;
+  background-color: @body-bg;
+  .shadow;
+}
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/ui/pages/article/article-page.less]: remove it

File: src/Uintra20/Content/uintra20/src/app/ui/pages/news/uintra-news-edit/uintra-news-edit-page.component.ts [hunk 5]
```diff
@@ -94,4 +97,17 @@ export class UintraNewsEditPage implements OnInit {
 
     return copyObject;
   }
+
+  canDeactivate(): Observable<boolean> | boolean {
+    if (this.hasDataChangedService.hasDataChanged) {
+      if(confirm('Are you sure?')) {
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/ui/pages/news/uintra-news-edit/uintra-news-edit-page.component.ts]: move to service. and replace it in all places

File: src/Uintra20/Content/uintra20/src/app/ui/panels/activity-create/sections/social-create/social-create.component.ts [hunk 6]
```diff
@@ -91,6 +98,16 @@ export class SocialCreateComponent implements OnInit {
     });
   }
 
+  onTagsChange(e) {
+    this.tags = e;
+  }
+
+  onDescriptionChange(e) {
+    debugger
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/ui/panels/activity-create/sections/social-create/social-create.component.ts]: remove it

File: src/Uintra20/Content/uintra20/src/app/feature/project/specific/activity/news-form/news-form.component.ts (hunk 2 above)
Comment (eugene-sahach) [reply src/Uintra20/Content/uintra20/src/app/feature/project/specific/activity/news-form/news-form.component.ts]: +

File: src/Uintra20/Content/uintra20/src/app/ui/pages/article/article-page.html (hunk 3 above)
Comment (eugene-sahach) [reply src/Uintra20/Content/uintra20/src/app/ui/pages/article/article-page.html]: +

File: src/Uintra20/Content/uintra20/src/app/ui/pages/article/article-page.less (hunk 4 above)
Comment (eugene-sahach) [reply src/Uintra20/Content/uintra20/src/app/ui/pages/article/article-page.less]: +

File: src/Uintra20/Content/uintra20/src/app/ui/panels/activity-create/sections/social-create/social-create.component.ts (hunk 6 above)
Comment (eugene-sahach) [reply src/Uintra20/Content/uintra20/src/app/ui/panels/activity-create/sections/social-create/social-create.component.ts]: +

File: src/Uintra20/Content/uintra20/src/app/ui/pages/news/uintra-news-edit/uintra-news-edit-page.component.ts (hunk 5 above)
Comment (eugene-sahach) [reply src/Uintra20/Content/uintra20/src/app/ui/pages/news/uintra-news-edit/uintra-news-edit-page.component.ts]: +

**Pull request merged**